apis/keda/v1alpha1/scaledobject_webhook.go

```diff
@@ -297,3 +310,115 @@ func verifyCPUMemoryScalers(incomingSo *ScaledObject, action string) error {
 	}
 	return nil
 }
+
+// ValidateComplexScalingLogic validates all combinations of given arguments
+// and their values
+func ValidateComplexScalingLogic(so *ScaledObject, specs []autoscalingv2.MetricSpec) (float64, autoscalingv2.MetricTargetType, error) {
```

Comment:
wasnt really sure where to put these functions. Putting them in `hpa.go` and then calling them in `apis/keda.../scaledobject_weebhook.go` creates a cyclical dependency.

Reply:
We have to place them here or in an external package used in both, but as you said, the hpa.go. isn't an option as it'd create a cyclic dependency

Reply:
putting it in a separate file could be better i think, i could also add the new external scaling option using grpc servers i have WIP for, as a second part of the parent PR for this PR.